### ISSUE TITLE
Support Multi-part message for POST

### ISSUE THREAD

**Issue by tobyweston** [simple-http/simple-http#7]

**Comment by tobyweston** [comment]
*Very* basic support is in. You can only have a single part in the multipart body and it's bias towards uploading a file. So you should be able to upload a single file. Use something like the following.

```java
File file = new File("src/test/resource/example-image.png");
HttpResponse response = HttpClients.anApacheClient().post(Url.url("http://ptsv2.com/t/rcari-1550250573/post"), new Multipart("example", file));
System.out.println("response = " + response);
```
which sends something like the following.

```
POST /rcari-1550250573/post
Content-Type: multipart/form-data; boundary=0Cm4Z1ek9YKQl0BLZ1a2tJHWKGOu3v-eH0xf58y

--pcJx2kexk6w2i5dv2MhXq38711TUKmwvhiith4t
Content-Disposition: form-data; name="example"; filename="example-image.png"
Content-Type: application/octet-stream

�PNG


IHDR|b�IDATx�cd�'�fIEND�B`�
--pcJx2kexk6w2i5dv2MhXq38711TUKmwvhiith4t--

```

* NB. Use http://ptsv2.com to setup a temporary "bin" to see what's uploaded and the message payload

**Comment by tobyweston** [comment]
@ralphcollett want to review things and maybe think about a release?

**Comment by tobyweston** [comment]
Included in v1.4
